datafusion/src/physical_plan/projection.rs

```diff
@@ -157,6 +158,42 @@ impl ExecutionPlan for ProjectionExec {
             }
         }
     }
+
+    fn statistics(&self) -> Statistics {
+        stats_projection(
+            self.input.statistics(),
+            self.expr.iter().map(|(e, _)| Arc::clone(e)),
+        )
+    }
+}
+
+fn stats_projection(
+    stats: Statistics,
+    exprs: impl ExactSizeIterator<Item = Arc<dyn PhysicalExpr>>,
+) -> Statistics {
+    let column_statistics = if let Some(input_col_stats) = stats.column_statistics {
+        let mut column_stat = Vec::with_capacity(exprs.len());
```

Comment:
Could be written using `exprs.iter().map(...).collect()` instead?

Reply:
you are right: it could, it should, it will 😄